youtube_dl/extractor/safari.py

```diff
@@ -238,3 +245,52 @@ def _real_extract(self, url):
         course_title = course_json['title']
 
         return self.playlist_result(entries, course_id, course_title)
+
+
+class SafariLearningPathIE(SafariBaseIE):
+    IE_NAME = 'safari:learning-path'
+    IE_DESC = 'safaribooksonline.com learning paths'
+
+    _VALID_URL = r'''(?x)
+                    https?://
+                        (?:
+                            (?:www\.)?safaribooksonline\.com/
+                            (?:learning-paths/[^/]+)
+                        )
+                        /(?P<id>[^/]+)
+                    '''
+
+    _TESTS = [{
+        'url': 'https://www.safaribooksonline.com/learning-paths/learning-path-python/9781788996396',
```

Comment:
```suggestion
        'url': 'https://learning.oreilly.com/learning-paths/learning-path-python/9781788996396',
```
I assume it would be good to include one link starting with "learning.oreilly.com" in tests.